glib-macros/src/object_interface_attribute.rs

```diff
@@ -209,29 +230,29 @@ fn register_object_interface_as_dynamic(
         }
     } else {
         // registers immediately the object interface as a dynamic type.
+
+        // name of the static variable to store the GLib type.
+        let gtype_status = format_ident!("{}_G_TYPE", self_ty.to_string().to_shouty_snake_case());
+
         quote! {
-            impl #self_ty {
-                /// Returns a mutable reference to the GLib type.
-                /// This is safe because the mutable reference guarantees that no other threads are concurrently accessing the atomic data.
-                #[inline]
-                fn get_type_mut() -> &'static mut #crate_ident::ffi::GType {
-                    static mut TYPE: ::std::sync::atomic::AtomicUsize = ::std::sync::atomic::AtomicUsize::new(#crate_ident::gobject_ffi::G_TYPE_INVALID);
-                    unsafe { TYPE.get_mut() }
-                }
+            /// The GLib type which can be safely shared between threads.
+            static #gtype_status: ::std::sync::atomic::AtomicUsize = ::std::sync::atomic::AtomicUsize::new(#crate_ident::gobject_ffi::G_TYPE_INVALID);
 
+            impl #self_ty {
                 /// Do nothing as the object interface has been registered on implementation load.
                 #[inline]
                 fn register_interface() -> #crate_ident::Type {
-                    unsafe { <#crate_ident::Type as #crate_ident::translate::FromGlib<#crate_ident::ffi::GType>>::from_glib(*Self::get_type_mut()) }
+                    let gtype = #gtype_status.load(::std::sync::atomic::Ordering::Relaxed);
```

Comment:
Is this somewhere protected also via a mutex? Otherwise it would probably be safer to use `SeqCst` here

Reply:
Yes, `Acquire` / `Release` seems to be enough here